x/config/config.go

```diff
@@ -26,6 +26,94 @@ import (
+}
+
+// NewDefaultConfigParser creates a [ConfigParser] with a set of default wrappers already registered.
+func NewDefaultConfigParser() *ConfigParser {
+	p := new(ConfigParser)
+
+	// Please keep the list in alphabetical order.
+
+	p.RegisterStreamDialerWrapper("override", wrapStreamDialerWithOverride)
+	p.RegisterPacketDialerWrapper("override", wrapPacketDialerWithOverride)
+
+	p.RegisterStreamDialerWrapper("socks5", wrapStreamDialerWithSOCKS5)
+	p.RegisterPacketDialerWrapper("socks5", func(baseDialer transport.PacketDialer, wrapConfig *url.URL) (transport.PacketDialer, error) {
+		return nil, errors.New("socks5 is not supported for PacketDialers")
+	})
+
+	p.RegisterStreamDialerWrapper("split", func(baseDialer transport.StreamDialer, wrapConfig *url.URL) (transport.StreamDialer, error) {
+		prefixBytesStr := wrapConfig.Opaque
+		prefixBytes, err := strconv.Atoi(prefixBytesStr)
+		if err != nil {
+			return nil, fmt.Errorf("prefixBytes is not a number: %v. Split config should be in split:<number> format", prefixBytesStr)
+		}
+		return split.NewStreamDialer(baseDialer, int64(prefixBytes))
+	})
+	p.RegisterPacketDialerWrapper("split", func(baseDialer transport.PacketDialer, wrapConfig *url.URL) (transport.PacketDialer, error) {
+		return nil, errors.New("split is not supported for PacketDialers")
+	})
+
+	p.RegisterStreamDialerWrapper("ss", wrapStreamDialerWithShadowsocks)
+	p.RegisterPacketDialerWrapper("ss", wrapPacketDialerWithShadowsocks)
+
+	p.RegisterStreamDialerWrapper("tls", wrapStreamDialerWithTLS)
+	p.RegisterPacketDialerWrapper("tls", func(baseDialer transport.PacketDialer, wrapConfig *url.URL) (transport.PacketDialer, error) {
+		return nil, errors.New("tls is not supported for PacketDialers")
+	})
+
+	p.RegisterStreamDialerWrapper("tlsfrag", func(baseDialer transport.StreamDialer, wrapConfig *url.URL) (transport.StreamDialer, error) {
+		lenStr := wrapConfig.Opaque
+		fixedLen, err := strconv.Atoi(lenStr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid tlsfrag option: %v. It should be in tlsfrag:<number> format", lenStr)
+		}
+		return tlsfrag.NewFixedLenStreamDialer(baseDialer, fixedLen)
+	})
+	p.RegisterPacketDialerWrapper("tlsfrag", func(baseDialer transport.PacketDialer, wrapConfig *url.URL) (transport.PacketDialer, error) {
+		return nil, errors.New("tlsfrag is not supported for PacketDialers")
+	})
+
+	return p
+}
+
+// WrapStreamDialerFunc wraps a [transport.StreamDialer] based on the wrapConfig.
+type WrapStreamDialerFunc func(dialer transport.StreamDialer, wrapConfig *url.URL) (transport.StreamDialer, error)
+
+// RegisterStreamDialerWrapper will register a wrapper for stream dialers under the given subtype.
+func (p *ConfigParser) RegisterStreamDialerWrapper(subtype string, wrapper WrapStreamDialerFunc) error {
+	if p.sdWrapers == nil {
+		p.sdWrapers = make(map[string]WrapStreamDialerFunc)
+	}
+
+	if _, found := p.sdWrapers[subtype]; found {
+		return fmt.Errorf("config parser %v added twice", subtype)
```

Comment:
*nit* Added more details for better debug purpose when the caller chose to output the error directly to `stderr`.

```suggestion
		return fmt.Errorf("config parser %v for StreamDialer added twice", subtype)
```